src/main/java/org/spongepowered/api/CatalogType.java

```diff
@@ -60,5 +64,13 @@
      * @return The human-readable name of this catalog type
      */
     String getName();
+    
+    @Override
+    default Text toText() {
```

Comment:
Again, the thing is, this requires that every `CatalogType` has a default implementation of a legible name that is presented for text, which as an object, is entirely immutable. presenting what a `BlockType` can be as a `Text` object is perfectly fine, but telling me that every `CatalogType` needs to be representable as `Text` objects with little to no configurable alternative is equivalent to ignoring that a `CatalogType` is `TextRepresentable` at all. What would make sense is if you exposed the ability to register `TextRepresenters` that could create a `Text` object for a `CatalogType` such that plugins can dictate how they want certain ones to be read in chat.

Default implementations or not, I seriously don't want to make this a contracted method for `CatalogType` in general as it's simply in the wrong place.